packages/backend/src/plugins/search.ts

```diff
@@ -14,27 +14,31 @@
  * limitations under the License.
  */
 import { useHotCleanup } from '@backstage/backend-common';
+import { DefaultCatalogCollator } from '@backstage/plugin-catalog-backend';
 import { createRouter } from '@backstage/plugin-search-backend';
+import { ElasticSearchSearchEngine } from '@backstage/plugin-search-backend-module-elasticsearch';
+import { PgSearchEngine } from '@backstage/plugin-search-backend-module-pg';
 import {
   IndexBuilder,
   LunrSearchEngine,
 } from '@backstage/plugin-search-backend-node';
-import { PluginEnvironment } from '../types';
-import { DefaultCatalogCollator } from '@backstage/plugin-catalog-backend';
 import { DefaultTechDocsCollator } from '@backstage/plugin-techdocs-backend';
-import { ElasticSearchSearchEngine } from '@backstage/plugin-search-backend-module-elasticsearch';
+import { PluginEnvironment } from '../types';
 
 export default async function createPlugin({
   logger,
   discovery,
   config,
+  database,
 }: PluginEnvironment) {
   // Initialize a connection to a search engine.
   const searchEngine = config.has('search.elasticsearch')
     ? await ElasticSearchSearchEngine.fromConfig({
         logger,
         config,
       })
+    : (await PgSearchEngine.supported(database))
+    ? await PgSearchEngine.from({ database })
```

Comment:
Maybe it would make sense to support postgres in the app template too, that way new adopters could get a search engine that they could use in production too. But that could also be something for a future PR.

Reply:
:100: I'll create a follow-up item to track.